### DISCUSSION TITLE
500 Internal Server Error

### DISCUSSION THREAD

**Discussion by cata008** [Azure/ipam#127] [General]
@DCMattyG I have deployed the IPAM solution using Azure Web Apps instead (one for the UI and one for the Engine), but I am running into some ugly errors. 

<img width="369" alt="image" src="https://github.com/Azure/ipam/assets/37331541/e7118488-a922-475d-940e-a2ceecb19013">

<img width="401" alt="image" src="https://github.com/Azure/ipam/assets/37331541/6dd77cb8-e1cf-4dea-bb16-7a312a39efe1">

If I click on any of the tabs - spaces, blocks, vnets etc - I end up with a loading screen. While checking the engine logs, I can see many errors, but I am not sure where to start looking for the problem.

<img width="590" alt="image" src="https://github.com/Azure/ipam/assets/37331541/297b75e7-af3a-4c1f-8460-6ffdf0638817">

<img width="531" alt="image" src="https://github.com/Azure/ipam/assets/37331541/6633aea4-f52d-4e9a-ab02-44b19cb883be">

The only thing that I did different is, I deployed the solution using an Application Admin role (I don't have a Global Admin role), but everything was successfully deployed nonetheless and all permissions seem to be in place. 

Any idea what I got wrong?

**Comment by DCMattyG** [comment]
Hey @cata008, the Global Admin requirement is really for the ability to grant [Admin Consent](https://learn.microsoft.com/en-us/azure/active-directory/manage-apps/grant-admin-consent) for the App Registration. For the Engine Service Principal, does it have *Reader* at the [Root Management Group](https://learn.microsoft.com/en-us/azure/governance/management-groups/overview#root-management-group-for-each-directory)?

I'm guessing that you're not using the provided ```deploy.ps1``` script, so how are you creating the App Registrations and assigning their permissions for your deployment? Are you doing that manually?

**Reply by cata008** [reply]
@DCMattyG Thanks so much for your quick reply. 

I have tried multiple things:
a. Deployed the solution in a multi-container app service using .\deploy.ps1 - still got the described behavior and no errors - maybe it is worth looking into error handling?
b. Started as "a." and then made the app service a single container app; deployed an additional app service for the engine - set all environment variables accordingly - the same behavior 
c. Just managed to get hold of a test tenant and with a Global Admin everything works 

This is how the app registrations look with an Application Admin role only:
ipam-ui-app
<img width="520" alt="image" src="https://github.com/Azure/ipam/assets/37331541/310fe7d9-f5cc-40bd-8162-c1dea8db7527">

ipam-ui-engine
<img width="507" alt="image" src="https://github.com/Azure/ipam/assets/37331541/83b64c2b-e3ee-4a03-ab2c-4d5f5af14ea5">

I compared this with the way it looks for the Global admin deployment and I do not spot a difference.  

And to answer to your questions, the engine spn has Reader on the root mg.

**Reply by DCMattyG** [reply]
Thank you @cata008, for the incredible level of detail. It seems you're doing everything as expected, so I'm curious to explore this with you on our upcoming Teams call together later this week.
